src/sentry/issues/ingest.py

```diff
@@ -13,6 +42,150 @@ def save_issue_occurrence(occurrence_data: IssueOccurrenceData, event: Event) ->
     # sure that this is somehow validated.
     occurrence.save(event.project_id)
 
-    # TODO: Create/update issue
+    # TODO: Pass release here
+    save_issue_from_occurrence(occurrence, event, None)
+    # TODO: Create group related releases here
     # TODO: Write occurrence and event eventstream
     return occurrence
+
+
+def process_occurrence_data(occurrence_data: IssueOccurrenceData) -> None:
+    # Hash fingerprints to make sure they're a consistent length
+    occurrence_data["fingerprint"] = [
+        md5(part.encode("utf-8")).hexdigest() for part in occurrence_data["fingerprint"]
+    ]
+
+
+class IssueArgs(TypedDict):
+    platform: Optional[str]
+    message: str
+    level: Optional[int]
+    culprit: str
+    last_seen: datetime
+    first_seen: datetime
+    active_at: datetime
+    type: int
+    data: OccurrenceMetadata
+    first_release: Optional[Release]
+
+
+def _create_issue_kwargs(
+    occurrence: IssueOccurrence, event: Event, release: Optional[Release]
+) -> IssueArgs:
+    kwargs: IssueArgs = {
+        "platform": event.platform,
+        # TODO: Figure out what message should be. Or maybe we just implement a platform event and
+        # define it in `search_message` there.
+        "message": event.search_message,
+        # TODO: Not sure what to put here
+        # "logger": job["logger_name"],
+        # TODO: Level override from occurrence?
+        "level": LOG_LEVELS_MAP.get(event.data["level"]),
+        "culprit": occurrence.subtitle,
+        "last_seen": event.datetime,
+        "first_seen": event.datetime,
+        "active_at": event.datetime,
+        "type": cast(int, occurrence.type.value),
+        "first_release": release,
+        "data": materialize_metadata(occurrence, event),
+    }
+    kwargs["data"]["last_received"] = event.datetime
+    return kwargs
+
+
+class OccurrenceMetadata(TypedDict):
+    type: str
+    culprit: str
+    metadata: Mapping[str, Any]
+    title: str
+    location: Optional[str]
+    last_received: datetime
+
+
+def materialize_metadata(occurrence: IssueOccurrence, event: Event) -> OccurrenceMetadata:
+    """
+    Returns the materialized metadata to be merged with issue.
+    """
+
+    event_type = get_event_type(event.data)
+    event_metadata = dict(event_type.get_metadata(event.data))
+    event_metadata = dict(event_metadata)
+    event_metadata["title"] = occurrence.issue_title
+
+    return {
+        "type": event_type.key,
+        # Not totally sure if this makes sense?
+        "culprit": occurrence.subtitle,
```

Comment:
i guess this makes sense. would probably be too much trouble to involve `IssueEvidence`. what do we show with our integrations currently that subtitle is replacing/substituting for?

Reply:
My memory is a bit vague on how the frontend displays things here on the issue list, but from what I remember `title` is the main title, `location` is the path and `culprit` is the subtitle. Tbh I'm not worried about getting this perfect right now, we can write this data here and modify it later if necessary. Comment was more a note to revisit it later if necessary.